src/Agent/NewRelic/Agent/Core/Core.csproj

```diff
@@ -33,21 +33,25 @@
 
   <ItemGroup>
     <PackageReference Include="Autofac" Version="4.5.0" />
-    <PackageReference Include="Google.Protobuf" Version="3.21.12" />
-    <PackageReference Include="Grpc.Tools" Version="2.28.1">
+    <PackageReference Include="Google.Protobuf" Version="3.25.2" />
+    <PackageReference Include="Grpc.Tools" Version="2.60.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Serilog" Version="3.0.1" />
+    <PackageReference Include="Serilog" Version="3.1.1" />
     <PackageReference Include="Serilog.Sinks.Async" Version="1.5.0" />
-    <PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
+    <PackageReference Include="Serilog.Sinks.Console" Version="5.0.1" />
     <PackageReference Include="Serilog.Sinks.Debug" Version="2.0.0" />
     <PackageReference Include="Serilog.Sinks.EventLog" Version="3.1.0" />
     <PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.0" />
```

Comment:
What required this dependency? I think that ILRepacking this dependency may have some unexpected side effects. This will result in the possibility of 2 different versions of diagnostic source being loaded into an app, which is often what we want. However, if there were assumptions about that instance of diagnostics source getting information from the rest of the application, it will not work. For example, Activity.Current will potentially have 2 different values because they are essentially different types because of the ILRepacking.

Reply:
Adding that dependency resolved a failing integration test around W3C TraceID tests -- it was specifically because `Activity.Current` had an unexpected value in the test. Once the package reference and ILRepack was added, the test passes.

Reply:
We should take a closer look at it during a team pairing session.

Reply:
Looks like the need for `DiagnosticSource` is driven by a [dependency on it from Serilog v3.1.1](https://www.nuget.org/packages/Serilog/3.1.1), for .NET Framework targets. So maybe I can scope it down to just the Framework build and (maybe) not need to ILRepack it?

Reply:
For now, I'm reverting this change and the Serilog change.